src/components/Banner/Banner.module.css

```diff
@@ -6,163 +6,203 @@
-  background-color: var(--eds-theme-color-neutral-subtle-background);
+  border-width: var(--eds-theme-border-width);
+  border-color: var(--border-light-color);
+
+  /* Vertical-specific styles */
+  padding: var(--eds-size-4);
+  text-align: center;
+  justify-items: center;
+  border-top-width: var(--eds-border-width-lg);
+  border-top-color: var(--border-dark-color); /* 3 */
 }
 
 /**
- * Banner success
- * 1) The thick left border operates as a foreground band of color,
- *    so foreground rather than border is used
+ * Banner informational icon
+ * 1) Icon color matches the thick border (left for horizontal and top for
+ *    vertical orientation)
  */
-.banner--success {
-  background: var(--eds-theme-color-utility-success-background);
-  border-color: var(--eds-theme-color-utility-success-border);
-  border-left-color: var(--eds-theme-color-utility-success-foreground); /* 1 */
-  color: var(--eds-theme-color-utility-success-foreground);
+.banner__icon {
+  color: var(--border-dark-color); /* 1 */
+  --icon-size-default: var(--eds-size-5);
 }
 
-/**
- * Banner warning
- */
-.banner--warning {
-  background: var(--eds-theme-color-utility-warning-background);
-  border-color: var(--eds-theme-color-utility-warning-border);
-  border-left-color: var(--eds-theme-color-utility-warning-foreground); /* 1 */
-  color: var(--eds-theme-color-utility-warning-foreground);
+.banner__textAndAction,
+.banner__textContent {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
 }
 
-/**
- * Banner error
- */
-.banner--error {
-  background: var(--eds-theme-color-utility-error-background);
-  border-color: var(--eds-theme-color-utility-error-border);
-  border-left-color: var(--eds-theme-color-utility-error-foreground); /* 1 */
-  color: var(--eds-theme-color-utility-error-foreground);
+.banner__textAndAction {
+  gap: var(--eds-size-2);
 }
 
-/**
- * Brand banner
- */
-.banner--brand {
-  background: var(--eds-color-brand-blue-100); /* TODO */
-  border-color: var(--eds-theme-color-primary-border);
-  border-left-color: var(--eds-theme-color-primary-foreground); /* 1 */
-  color: var(--eds-theme-color-primary-foreground);
+.banner__textContent {
+  gap: var(--eds-size-1);
+}
+
+.banner__action {
+  justify-self: center;
+  align-self: center;
 }
 
 /**
- * Banner informational icon
- * 1) Selector doubled up to bump specificity to not interact with .icon.
- *    See code guidelines for more info.
+ * Banner close button
+ * 1) Button used to dismiss dismissable banners
+ * 2) Match close button icon color to the banner text color
  */
-.banner__icon.banner__icon {
-  margin-right: var(--eds-size-2);
-  fill: var(--eds-theme-color-neutral-medium-foreground);
-  width: var(--eds-size-3);
-  height: var(--eds-size-3);
+.banner__close-btn.banner__close-btn {
+  position: absolute;
+  top: var(--eds-size-half);
+  right: var(--eds-size-half);
+  /* TODO: handle hover styles */
+  color: inherit; /* 2 */
```

Comment:
The `Button` is being changed in https://github.com/chanzuckerberg/edu-design-system/pull/977 to accommodate this need. I could stack this PR on top of that one in order to use those styles, but I've been getting tired of rebasing from complex PRs, so I'm instead going to wait until that one merges.